### PR TITLE
refactor: make sync agg index default

### PR DESCRIPTION
**Pull request by ariesdevil**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Make sync agg index default, change the optional `sync` to `async`

Fixes #[Link the issue here]

## Tests

- [x] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14140)

### CODE REVIEW

File: src/query/storages/fuse/src/operations/agg_index_sink.rs [hunk 1]
```diff
@@ -68,12 +77,36 @@ impl AggIndexSink {
...
             location_data: HashMap::new(),
             blocks: vec![],
+            max_memory_usage,
         });
 
         Ok(ProcessorPtr::create(sinker))
     }
 
-    fn process_block(&mut self, block: &mut DataBlock) {
+    fn check_memory_limit(&self, block: &DataBlock) -> Result<()> {
```
Comment (BohuTANG) [inline src/query/storages/fuse/src/operations/agg_index_sink.rs:86]: We have the soft oom by default, does it also working?

Comment (ariesdevil) [reply src/query/storages/fuse/src/operations/agg_index_sink.rs:86]: Yup it works, I have add tests here: https://github.com/datafuselabs/databend/pull/14140/files#diff-4001267ac227fed7b868cb28249ed3020c48bf6c843cb120e213ac816cdbd092

Comment (BohuTANG) [reply src/query/storages/fuse/src/operations/agg_index_sink.rs:86]: So we don't need `check_memory_limit`?

Comment (ariesdevil) [reply src/query/storages/fuse/src/operations/agg_index_sink.rs:86]: If we don't check memory here, how do we return the corresponding error message?

Comment (BohuTANG) [reply src/query/storages/fuse/src/operations/agg_index_sink.rs:86]: No need return, we have many places to return soft oom.

Comment (BohuTANG) [reply src/query/storages/fuse/src/operations/agg_index_sink.rs:86]: This is the soft oom error:
image

**Review by RinChanNOWWW — APPROVED** [review]

**Pull request merged**